### ISSUE TITLE
RemoveUnusedTerms does not work on class/object

### ISSUE THREAD

**Issue by MasseGuillaume** [scalacenter/scalafix#626]
RemoveUnusedTerms does not produce a diff on stats:

```scala
class A {
  private val a = 1
}
object B {
  private val a = 1
}
package object C {
  private val a = 1
}
trait A {
  private val a = 1
}
```

**Comment by joan38** [comment]
This can be kept open since #723 didn't solve this issue.
The issue is in SemanticDB not forwarding the compiler warning message.
@olafurpg Do you have an idea of how we can fix this?

**Comment by olafurpg** [comment]
We can open a ticket in scalameta/scalameta with the reproduction in https://github.com/scalacenter/scalafix/pull/723#issuecomment-387017310

I'm sorry I won't have time to investigate this much in the coming weeks as I need to prioritize #704 and it seems it will require a fairly deep refactoring.

**Comment by joan38** [comment]
Don't worry I'll open the issue on Scalameta and we can see later.

**Comment by kubukoz** [comment]
The unused val warning is actually there for the last case, btw

**Comment by joan38** [comment]
This should be solved with the latest Scalameta

**Comment by gabro** [comment]
So, I've tried to tackle this with Scalameta 4 and here's what I found.

Given the 4 cases from the OP, the trait one works correctly in Scala 2.12, whereas the other 3 are still not working with the current implementation of `RemoveUnused` and none of them is working in Scala 2.11.

This is due to the positions of the warning. In Scala 2.12:

https://github.com/scalameta/scalameta/blob/7cdcba551547d72a4ac7ec3063e86ef2c861d574/tests/jvm/src/test/scala-2.12/scala/meta/tests/semanticdb/TargetedSuite2_12.scala#L21-L25

the reported position is precisely on the identifier being defined, except for the trait case, where the warning spans on the whole statement.
